api/server/runner_fninvoke.go

```diff
@@ -103,11 +108,36 @@ func (s *Server) fnInvoke(resp http.ResponseWriter, req *http.Request, app *mode
 	writer.Header().Add("Fn-Call-Id", call.Model().ID) // XXX(reed): move to before Submit when adding streaming
 
 	// buffered response writer traps status (so we can add headers), we need to write it still
-	if writer.status > 0 {
-		resp.WriteHeader(writer.status)
+	if writer.Status() > 0 {
+		resp.WriteHeader(writer.Status())
+	}
+
+	if isDetached {
+		return nil
 	}
 
 	io.Copy(resp, buf)
 	bufPool.Put(buf) // at this point, submit returned without timing out, so we can re-use this one
 	return nil
 }
+
+func getCallOptions(req *http.Request, app *models.App, fn *models.Fn, trig *models.Trigger, rw http.ResponseWriter) []agent.CallOpt {
+	var opts []agent.CallOpt
+	if req.Header.Get("Fn-Invoke-Type") == models.TypeDetached {
+		opts = []agent.CallOpt{
+			agent.WithWriter(rw), // XXX (reed): order matters [for now]
+			agent.FromHTTPFnRequest(app, fn, req),
+			agent.InvokeDetached(),
+		}
+	} else {
+		opts = []agent.CallOpt{
+			agent.WithWriter(rw), // XXX (reed): order matters [for now]
+			agent.FromHTTPFnRequest(app, fn, req),
+		}
+	}
```

Comment:
nit: this could be 5 loc instead of 11, using append when detached, and less risk of dupe code mistakes

Reply:
done